include/verilated_random.cpp

```diff
@@ -404,7 +374,13 @@ bool VlRandomizer::next(VlRNG& rngr) {
     f << "(define-fun __Vbv ((b Bool)) (_ BitVec 1) (ite b #b1 #b0))\n";
     f << "(define-fun __Vbool ((v (_ BitVec 1))) Bool (= #b1 v))\n";
     for (const auto& var : m_vars) {
-        f << "(declare-fun " << var.second->name() << " () ";
+        if (var.second->dimension() > 0) {
+            auto m_arr_vars_ptr
+                = std::make_shared<const std::map<std::string, std::shared_ptr<const ArrayInfo>>>(
+                    m_arr_vars);
+            var.second->setArrayInfo(m_arr_vars_ptr);
```

Comment:
```suggestion
            auto arrVarsp
                = std::make_shared<ArrayInfoMap>(m_arr_vars);
            var.second->setArrayInfo(arrVarsp);
```

Similar below too. And make a "using" typedef for ArrayInfoMap, and use that in the .h declaration also.